sycl/source/detail/usm/opencl_shim.cpp

```diff
@@ -0,0 +1,302 @@
+    return 
+        clEnqueueSVMMemcpy(queue, blocking, dst_ptr, src_ptr, size,
+                           num_events_in_wait_list, event_wait_list, event);
+  }
+
+  return CL_INVALID_OPERATION;
+}
+
+static clEnqueueMigrateMemINTEL_fn pfn_clEnqueueMigrateMemINTEL = NULL;
+CL_API_ENTRY cl_int CL_API_CALL clEnqueueMigrateMemINTEL(
+    cl_command_queue queue, const void *ptr, size_t size,
+    cl_mem_migration_flags flags, cl_uint num_events_in_wait_list,
+    const cl_event *event_wait_list, cl_event *event) {
+  if (pfn_clEnqueueMigrateMemINTEL) {
+    return pfn_clEnqueueMigrateMemINTEL(
+      queue,
+      ptr,
+      size,
+      flags,
+      num_events_in_wait_list,
+      event_wait_list,
+      event);
+  }
+  else if (GetCLUSM()) {
+    // We could check for OpenCL 2.1 and call the SVM migrate
+    // functions, but for now we'll just enqueue a marker.
+#if 0
+    retVal = clEnqueueSVMMigrateMem(
+      queue,
+      1,
+      &ptr,
+      &size,
+      flags,
+      num_events_in_wait_list,
+      event_wait_list,
+      event );
+#else
+    return clEnqueueMarkerWithWaitList(queue, num_events_in_wait_list,
+                                       event_wait_list, event);
+#endif
+  }
+
+  return CL_INVALID_OPERATION;
+}
+
+static clEnqueueMemAdviseINTEL_fn pfn_clEnqueueMemAdviseINTEL = NULL;
+CL_API_ENTRY cl_int CL_API_CALL clEnqueueMemAdviseINTEL(
+    cl_command_queue queue, const void *ptr, size_t size,
+    cl_mem_advice_intel advice, cl_uint num_events_in_wait_list,
+    const cl_event *event_wait_list, cl_event *event) {
+  if (pfn_clEnqueueMemAdviseINTEL) {
+    return pfn_clEnqueueMemAdviseINTEL(
+      queue,
+      ptr,
+      size,
+      advice,
+      num_events_in_wait_list,
+      event_wait_list,
+      event);
+  }
+  else if (GetCLUSM()) {
+    // TODO: What should we do here?
+    // This isn't really supported yet.
+    // Advice is typically safe to ignore,
+    //  so a NOP will do.
+    return clEnqueueMarkerWithWaitList(queue, num_events_in_wait_list,
+                                       event_wait_list, event);
+  }
+
+  return CL_INVALID_OPERATION;
+}
+
+#ifdef __cplusplus
+namespace cl {
+namespace sycl {
+namespace detail {
+namespace cliext {
+#endif
+
+#define GET_EXTENSION( _funcname )                                      \
+    pfn_ ## _funcname = ( _funcname ## _fn )                            \
+        clGetExtensionFunctionAddressForPlatform(platform, #_funcname);
+
+void initializeExtensions( cl_platform_id platform )
+{
+  GET_EXTENSION( clHostMemAllocINTEL );
+  GET_EXTENSION( clDeviceMemAllocINTEL );
+  GET_EXTENSION( clSharedMemAllocINTEL );
+  GET_EXTENSION( clMemFreeINTEL );
+  GET_EXTENSION( clGetMemAllocInfoINTEL );
+  GET_EXTENSION( clSetKernelArgMemPointerINTEL );
+  GET_EXTENSION( clEnqueueMemsetINTEL );
+  GET_EXTENSION( clEnqueueMemcpyINTEL );
+  GET_EXTENSION( clEnqueueMigrateMemINTEL );
+  GET_EXTENSION( clEnqueueMemAdviseINTEL );
+}
+
+#ifdef __cplusplus
+}
```

Comment:
It would be great to add a comment for each `}` with the scope name.
```c++
} // namespace cliext
} // namespace detail
} // namespace sycl
} // namespace cl
```